client/me/purchases/manage-purchase/payment-info-block.tsx

```diff
@@ -108,6 +108,16 @@ export default function PaymentInfoBlock( {
 		);
 	}
 
+	if ( ! isRechargeable( purchase ) && hasPaymentMethod( purchase ) ) {
```

Comment:
Now that I think about it, this condition might also benefit from adding `purchase.isAutoRenewEnabled &&`

Reply:
Thank you, I have added `&&purchase.isAutoRenewEnabled` to both the purchases and the purchase item pages.